packages/core/src/render3/util/discovery_utils.ts

```diff
@@ -203,6 +218,61 @@ export function getDirectives(element: Element): {}[] {
   return context.directives === null ? [] : [...context.directives];
 }
 
+/**
+ * Metadata for a directive developers can use for debugging purposes or tooling.
+ */
+export interface DirectiveMetadata {
+  inputs: Record<string, string>;
+  outputs: Record<string, string>;
+}
```

Comment:
I think we should add `Debug` to the interface names, so they are not confused with non-debug interfaces. Also it'd be great to mention that it's a partial metadata and only a certain subset of fields is avaialble.
```suggestion
/**
 * Partial metadata for a given directive instance.
 * This information might be useful for debugging purposes or tooling.
 * Currently only `inputs` and `outputs` metadata is available.
 */
export interface DirectiveDebugMetadata {
  inputs: Record<string, string>;
  outputs: Record<string, string>;
}
```